### PR TITLE
Re-add clear notifications button

### LINKED ISSUE #3671: Clear all Notifications Button Missing from Notifications Column Header
The clear all notifications button (eraser/trashcan) is missing on versions after the adjustable columns were added (1.4.0+).

* * * *

- [Yes] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

### PR DESCRIPTION
**Pull request by ykzts**
Resolve #3671

### CODE REVIEW

File: app/javascript/mastodon/features/notifications/components/column_settings.js [hunk 1]
```diff
@@ -66,4 +87,4 @@ class ColumnSettings extends React.PureComponent {
 
 }
 
-export default injectIntl(ColumnSettings);
+export default connect()(injectIntl(ColumnSettings));
```
Comment (Gargron) [inline app/javascript/mastodon/features/notifications/components/column_settings.js]: This component is already wrapped in ColumnSettingsContainer (that's where settings, onChange and onSave come from) so connecting it again directly here is undesired. Adjust the container component instead.

File: app/javascript/mastodon/features/notifications/components/clear_column_button.js [hunk 2]
```diff
@@ -17,9 +13,15 @@ class ClearColumnButton extends React.Component {
...
     return (
-      <div role='button' title={intl.formatMessage(messages.clear)} className='column-icon column-icon-clear' tabIndex='0' onClick={this.props.onClick}>
+      <button
+        className='text-btn column-header__setting-btn'
+        tabIndex='0'
+        onClick={this.props.onClick}
+      >
         <i className='fa fa-eraser' />
-      </div>
+        {' '}
```
Comment (Gargron) [inline app/javascript/mastodon/features/notifications/components/clear_column_button.js]: `<i />` and `<FormattedMessage />` can be on the same line with a space between them

**Review by Gargron — CHANGES_REQUESTED** [review]

File: app/javascript/mastodon/features/notifications/components/clear_column_button.js (hunk 2 above)
Comment (ykzts) [inline app/javascript/mastodon/features/notifications/components/clear_column_button.js]: fix d94ca75

File: app/javascript/mastodon/features/notifications/components/column_settings.js (hunk 1 above)
Comment (ykzts) [inline app/javascript/mastodon/features/notifications/components/column_settings.js]: fix 973e883

**Review by Gargron — APPROVED** [review]

**Pull request merged**